.github/workflows/commands.yml

```diff
@@ -20,6 +20,7 @@ jobs:
     runs-on: ubuntu-22.04
     permissions:
       pull-requests: write
+      contents: write
```

Comment:
cc @joycebrum - seems like this requires repo write to restart workflows

Reply:
you sure it needs contents: write? looking at the doc (https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#actions) it seems that action write would be the one needed: [POST /repos/{owner}/{repo}/actions/runs/{run_id}/rerun](https://docs.github.com/en/rest/actions#re-run-a-workflow)

But since I don't know in depth what envoyproxy/toolshed/gh-actions/retest does, I can be totally wrong 😅 .

Reply:
>  you sure it needs contents: write?

not sure - i was following this

> Who can use this feature
> 
> People with write permissions to a repository can re-run workflows in the repository.

https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs

i also found various discussions about it needing write, and updates to the docs, but i wasnt able to find anything about permissions required in workflows to trigger workflows

Reply:
> ...  don't know in depth what jpsim/retest or envoyproxy/toolshed/gh-actions/retest does ...

it restarts failed github workflows

Reply:
ill update to use `actions: write` and we can see if it works - thanks!

Reply:
> ... updates to the docs ...

apologies for confusion - i meant the github docs were updated to reflect the need for write perms

Reply:
I've tested and it really needs actions: write permission https://github.com/joycebrum/envoy/actions/runs/5017672594

It at least ran fine for me in the fork only with these two permissions.

Reply:
brilliant, thanks!

Reply:
i think its not triggering any actual workflow reruns

Reply:
as in they arent failed currently

Reply:
https://github.com/envoyproxy/envoy/pull/27487#issuecomment-1553571431

it worked!

brilliant, thanks again @joycebrum for your help